pkg/operator/ceph/file/filesystem.go

```diff
@@ -211,6 +211,13 @@ func (f *Filesystem) doFilesystemCreate(context *clusterd.Context, clusterInfo *
 		if err := SetPoolSize(f, context, clusterInfo, spec, cluster); err != nil {
 			return errors.Wrap(err, "failed to set pools size")
 		}
+
+		dataPoolNames := generateDataPoolNames(f, spec)
```

Comment:
How about factoring out lines 200-221 to a new method for updating an existing filesystem? This conditional is getting long.